filters/filters.go

```diff
@@ -31,13 +31,17 @@ const (
 	BackendRatelimit = "backend:ratelimit"
 )
 
-// Context object providing state and information that is unique to a request.
+// FilterContext object providing state and information that is unique to a request.
 type FilterContext interface {
 	// The response writer object belonging to the incoming request. Used by
 	// filters that handle the requests themselves.
 	// Deprecated: use Response() or Serve()
 	ResponseWriter() http.ResponseWriter
 
+	// ResponseController returns the http.ResponseController
+	// attached to the current request processing.
+	ResponseController() *http.ResponseController
```

Comment:
We deprecated ResponseWriter() but adding ResponseController() that exposes low-level http details.
Alternative could be to implement it like we implemented `backendTimeout` - store timeouts in the statebag and apply in proxy before doing backed request.

Reply:
The problem is that these (read and write timeout) are not exposed.

Reply:
Yeah, I mean create ResponseController in proxy.go around here instead of exposing it from the context:
https://github.com/zalando/skipper/blob/3c60b4ee71d0b18a36f889b26904279e0aa1e33c/proxy/proxy.go#L1124-L1127

Reply:
yes, did that an reused the time.Now value from just there.